tests/Aspire.Cli.Tests/Utils/CliTestHelper.cs

```diff
@@ -27,13 +28,72 @@ public static IServiceCollection CreateServiceCollection(ITestOutputHelper outpu
 
         var services = new ServiceCollection();
 
-        var configuration = new ConfigurationBuilder().Build();
+        // Build configuration for testing
+        var configBuilder = new ConfigurationBuilder();
+        
+        // If explicit settings content is provided, use that instead of file discovery
+        if (!string.IsNullOrEmpty(options.LocalSettingsContent) || !string.IsNullOrEmpty(options.GlobalSettingsContent))
+        {
+            // Add local settings first (if provided)
+            if (!string.IsNullOrEmpty(options.LocalSettingsContent))
+            {
+                var localBytes = Encoding.UTF8.GetBytes(options.LocalSettingsContent);
+                var localStream = new MemoryStream(localBytes);
+                configBuilder.AddJsonStream(localStream);
+            }
+
+            // Then add global settings (if provided) - this will override local settings
+            if (!string.IsNullOrEmpty(options.GlobalSettingsContent))
+            {
+                var globalBytes = Encoding.UTF8.GetBytes(options.GlobalSettingsContent);
+                var globalStream = new MemoryStream(globalBytes);
+                configBuilder.AddJsonStream(globalStream);
+            }
+        }
+        else
+        {
+            // Fallback to original file discovery behavior for backward compatibility
+            // Find the nearest local settings file
+            var currentDirectory = new DirectoryInfo(Directory.GetCurrentDirectory());
```

Comment:
@coilot CreateServiceCollection should not attempt to find the files at all. Just use the values from options.